.github/workflows/main.yml

```diff
@@ -58,8 +58,8 @@ jobs:
           - os: "ubuntu-latest"
             build-type: Debug
             self-host: true
-            extra-cmake-flags: "-DSNMALLOC_MEMCPY_BOUNDS=ON -DSNMALLOC_CHECK_LOADS=ON"
-            dependencies: "sudo apt install ninja-build"
+            extra-cmake-flags: "-DSNMALLOC_MEMCPY_BOUNDS=ON -DSNMALLOC_CHECK_LOADS=ON -DCMAKE_CXX_COMPILER=g++-10"
```

Comment:
When you change us to using GCC it was so that people using the defaults on Ubuntu would get a well tested implementation.  I'm little concerned this will just mean we have many issues raised in the future.

Reply:
We can add something to CMake to warn that gcc 9 miscompiles snmalloc.  My experience with Ubuntu 18.04 is that it ships a very buggy compiler.  I've wasted time elsewhere with code that it miscompiles.  I'm happy if it can be made to work but it seems to just generate invalid binaries and I don't see how to fix that.  GCC 10 doesn't have this problem and is in the apt repo.

After some more changes, gcc 9.4 seems to work again.  So, uh, yay?  Reverted.